packages/guides/src/Compiler/Passes/FooternoteNumberedTargetPass.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\Passes;
+
+use phpDocumentor\Guides\Compiler\CompilerPass;
+use phpDocumentor\Guides\Meta\FootnoteTarget;
+use phpDocumentor\Guides\Metas;
+use phpDocumentor\Guides\Nodes\CompoundNode;
+use phpDocumentor\Guides\Nodes\FootnoteNode;
+use phpDocumentor\Guides\Nodes\Node;
+
+/**
+ * Resolves the hyperlink target for each section in the document.
+ *
+ * This follows the reStructuredText rules as outlined in:
+ * https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#implicit-hyperlink-targets
+ */
+class FooternoteNumberedTargetPass implements CompilerPass
```

Comment:
I'm wondering why you didn't combine the footnote passes, they look the same except for 1 statement. `$child->getNumber()` Looks like you could have done this in a single class.

Reply:
As is mentioned in the comment, numbered footnotes have to be dealt with before the named footnotes. Named footnotes get automatic numbers assigned but numbered footnotes take precedence. without passing all numbered footnotes first we cannot know which numbers are still open